### PR TITLE
OOP school library: decorate a class

### PR DESCRIPTION
**Pull request by fpsapc**
- Project requirements
Check the usage of [Decorator pattern in Ruby](https://refactoring.guru/design-patterns/decorator/ruby/example#example-0)
Think about how you can use two decorators in order to capitalize and trim people's names.
In the following steps you will get instructions for implementing the Decorator design pattern for this case.
- Interface
  - Create a class Nameable.
  - Implement a method called correct_name that will raise a NotImplementedError.
- Turn your Person class to Nameable
  - Make sure that your Person class inherits from Nameable
  - Make sure that this class has a method correct_name implemented. It should simply return the name attribute.
- Prepare base Decorator
  - Make sure that it inherits from Nameable.
  - In the constructor assign a nameable object from params to an instance variable.
  - Implement the correct_name method that returns the result of the correct_name method of the @nameable.
- Prepare CapitalizeDecorator and TrimmerDecorator
  - For the CapitalizeDecorator:
  - Create a class that inherits from the base Decorator class.
  - Implement a method correct_name that capitalizes the output of @nameable.correct_name.
- For the TrimmerDecorator:
  - Create a class that inherits from the base Decorator class.
  - Implement a method correct_name that makes sure that the output of @nameable.correct_name has a maximum of 10 
  characters. If it's longer it should trim the word.

- See your decorators in action
\Try the following code and check if you managed to decorate your person:

person = Person.new(22, 'maximilianus')
  person.correct_name
  capitalizedPerson = CapitalizeDecorator.new(person)
  capitalizedPerson.correct_name
  capitalizedTrimmedPerson = TrimmerDecorator.new(capitalizedPerson)
  capitalizedTrimmedPerson.correct_name

### CODE REVIEW

File: decorator.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+class Decorator
+  def initialize(nameable)
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
```
Comment (abdulhamiid) [inline decorator.rb]: According to the project requirements, kindly ensure that the `Decorator` class inherits from Namable 🛠👍

Requirements screenshot

image

```ruby
class Decorator < Nameable
```
When we use the `<` symbol to signify that the `Decorator` class is inheriting from the `Nameable` class. This means that all of the methods in the `Nameable` class are available to the `Decorator` class for use. 💪

**Review by abdulhamiid — CHANGES_REQUESTED** [review]
# Status: Required Changes :recycle:
Hi @fpsapc 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there! :running_man: :runner: 
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights 🌟

- ✔️ Implement composition 
- ✔️ Run a program using the command line.

## Required Changes ♻️
_Please kindly check the comments under the review._

## Optional suggestions 🤷‍♂️
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._


Cheers, and Happy coding!👏👏👏

## Important Notes 📝
- Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
- Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md
Comment (KanzaTahreem) [inline README.md:1]: - [OPTIONAL] Please consider adding **_Show your support_** and **_Acknowledgment_** sections in README for making it more professional ⛏🤓

**Review by KanzaTahreem — APPROVED** [review]
# Approved 🙌

Hi @fpsapc 🙋‍♀️

Great work implementing the suggested changes 💪 **_Your project is COMPLETE! There is nothing else to say other than... it's time to merge it! 🚀 Congratulations!_**



## Highlights 🎯
There are some excellent points that you have made perfectly. Keep up the great work! 👏

-  ✅ Implemented decorator, perfect!
-  ✅ Great work using the right Git Flow 
-  ✅ Linters are working properly, great job! 

## Optional suggestions ❕

_See the comments below_ 

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👨‍💻💻

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**